modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -2094,15 +2092,20 @@ def _build_index_cache(self):
                         RangeIndex, data=range(obj.num_rows)
                     )
                 else:
-                    index_at = obj.drop([f"F_{col}" for col in self.columns])
+                    col_names = [
+                        n for n in obj.column_names if not n.startswith(IDX_COL_NAME)
+                    ]
+                    index_at = obj.drop(col_names)
                     index_df = index_at.to_pandas()
-                    index_df.set_index(
-                        [f"F_{col}" for col in self._index_cols], inplace=True
-                    )
-                    index_df.index.rename(
-                        self._index_names(self._index_cols), inplace=True
-                    )
-                    self._index_cache = index_df.index
+                    index_df.set_index(self._index_cols, inplace=True)
+                    idx = index_df.index
+                    idx.rename(self._index_names(self._index_cols), inplace=True)
+                    if (
+                        isinstance(idx, (pd.DatetimeIndex, pd.TimedeltaIndex))
+                        and len(idx) > 3
```

Comment:
why is `> 3` needed? let's add an in-code comment about this

Reply:
Done